src/i18n/index.spec.js

```diff
@@ -0,0 +1,138 @@
+import i18next from 'i18next'
+import { i18nOptions } from '@/i18n/index'
+
+import _cloneDeep from 'lodash/cloneDeep'
+
+// HERE we can overwrite the translations for easier unit-testing
+const translation = {
+  simple: 'simple string',
+  interpolated: 'The text with some {{stuff}}',
+  withFormattedNumber: 'Congrats, it is your {{count, number}} visit!',
+  withFormattedCurrency: 'Your balance is {{amount, money}}',
+  withFormattedOrderNumber: 'You are in the {{place, order_number}} place',
+  withFormattedDateFuture: 'The offer ends {{when, date}}',
+  withFormattedDatePast: 'The offer ended {{when, date}}',
+  withFormattedInteger: 'John bought {{amount, integer}} apples'
+}
+
+describe('the i18n is properly configured', () => {
+  beforeEach(() => {
+    const options = _cloneDeep(i18nOptions)
+    options.resources.en.translation = translation
+    options.debug = false // Set to true, if something is not working
+    i18next.init(options)
+  })
+
+  it('init test', () => {
+    expect(i18next.t('simple')).to.equal('simple string')
+  })
+
+  it('interpolates properly', () => {
+    const result = i18next.t('interpolated', {
+      stuff: 'awesome stuff'
+    })
+    expect(result).to.equal('The text with some awesome stuff')
+  })
+
+  describe('formats the number', () => {
+    it('formats the simple number', () => {
+      const result = i18next.t('withFormattedNumber', {
+        count: '1000'
+      })
+      expect(result).to.equal('Congrats, it is your 1,000 visit!')
+    })
+
+    it('formats the custom currency', () => {
+      const result = i18next.t('withFormattedCurrency', {
+        amount: {
+          value: '1020',
+          currency: 'ETH'
+        }
+      })
+      expect(result).to.equal('Your balance is 1,020 ETH')
+    })
+
+    it('formats the custom currency with floating point', () => {
+      const result = i18next.t('withFormattedCurrency', {
+        amount: {
+          value: '5000.560010',
+          currency: 'QTK'
+        }
+      })
+      expect(result).to.equal('Your balance is 5,000.56001 QTK')
+    })
+
+    it('formats the preset currency', () => {
+      const result = i18next.t('withFormattedCurrency', {
+        amount: {
+          value: '1002.500000',
+          currency: 'USD'
+        }
+      })
+      expect(result).to.equal('Your balance is $1,002.5')
+    })
+
+    it('formats the currency without the provided code', () => {
+      const result = i18next.t('withFormattedCurrency', {
+        amount: '2010.500000'
+      })
+
+      expect(result).to.equal('Your balance is 2,010.5')
```

Comment:
Also for such functions that could return many results by performing similar operation I’d recommend using table-driven testing. For example, you test transformation of numbers like  `2010.500000` and you ensured that it returns `2,010.5`. But you cannot be sure that `20102030` will be formatted as `20,102,030`. In such cases it’s recommended to cover as much test-cases as you can to be ensured that at least known-for-you numbers are formatted well.